src/MSONUtility.h

```diff
@@ -418,6 +420,54 @@ namespace mson {
             propertyName.literal = subject;
         }
     }
+
+    /**
+     * \brief Add a dependency to the dependency list of the dependents while checking for circular references
+     *
+     * \param node Current markdown node iterator
+     * \param pd Section parser data
+     * \param dependency The named type which should be added to the dependency list
+     * \param dependent The named type to which the dependency should be added
+     * \param report Parse result report
+     */
+    inline void addDependency(const mdp::MarkdownNodeIterator& node,
+                              snowcrash::SectionParserData& pd,
+                              const mson::Literal dependency,
+                              const mson::Literal dependent,
+                              snowcrash::Report& report) {
+
+        std::set<mson::Literal> dependencyDeps = pd.namedTypeDependencyTable[dependency];
+
+        // First, check if it is circular reference between them
+        if (dependent == dependency ||
+            dependencyDeps.find(dependent) != dependencyDeps.end()) {
+
+            // ERR: Dependency named type circular references itself
+            std::stringstream ss;
+            ss << "base type '" << dependent << "' circularly referencing itself is not allowed";
```

Comment:
again see my comment on this message above, also if they are the same error manifested by the same message a reuse of the message would be appropriate